osu.Game/Database/StandardisedScoreMigrationTools.cs

```diff
@@ -185,6 +187,39 @@ public static long GetOldStandardised(ScoreInfo score)
             return (long)Math.Round((1000000 * (accuracyPortion * accuracyScore + (1 - accuracyPortion) * comboScore) + bonusScore) * modMultiplier);
         }
 
+        public static long ChangeOsuComboRatio(ScoreInfo score, double oldComboPortion, double newComboPortion)
+        {
+            if (score.RulesetID != 0)
+                throw new ArgumentException($"{nameof(score)} must be an osu! ruleset score");
+
+            // Assume the incoming score already has a total score in the newest standardised calculation method.
+            double totalScore = score.TotalScore;
+
+            // Undo mod multiplier to leave us with combo and accuracy portions.
+            double scoreMultiplier = 1;
+            foreach (var m in score.Mods)
+                scoreMultiplier *= m.ScoreMultiplier;
+
+            totalScore /= scoreMultiplier;
```

Comment:
In testing with the [database provided](https://github.com/ppy/osu/pull/23892#issuecomment-1588033153) for the last migration, this can actually fail due to:

https://github.com/ppy/osu/blob/23aa1752c331372fc4fd8b6a1688197d9f09da7d/osu.Game/Rulesets/Mods/UnknownMod.cs#L18

Score has unknown mod => `scoreMultiplier` becomes zero => `totalScore` becomes `double.NaN` and then `long.MinValue` at the cast at the end of the method.